File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) 
{
    vector<int> vec(s.length(),0);
    
    for(int i=0; i<s.length(); i++)
    {
        int currIdx = i;
        for(int j=0; j<currIdx; j++)
        {
            if(s[i] == s[j]) { vec[i] = i-j; }
        }
    }
    
    for(int i=0; i<vec.size(); i++)
    {
        if(vec[i] == 0) { vec[i] = -1; }
    }
    return vec;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cs
using System;

public class Solution 
{
    public int[] solution(string s) 
    {
        int[] result = new int[s.Length];
        for(int i=0; i<s.Length; i++)
        {
            char c = s[i];
            bool isMatch = false;
            for(int j=i-1; j>=0; j--)
            {
                if(c == s[j])
                {
                    result[i] = i - j;
                    isMatch = true;
                    break;
                }
            }
            if(!isMatch)
            {
                result[i] = -1;
            }
        }
        return result;
    }
}